homeassistant/components/home_plus_control/switch.py

```diff
@@ -0,0 +1,179 @@
+
+
+class HomeControlSwitchEntity(CoordinatorEntity, SwitchEntity):
+    """Entity that represents a Legrand Home+ Control switch.
+
+    It extends the HomeAssistant-provided classes of the CoordinatorEntity and the SwitchEntity.
+
+    The CoordinatorEntity class provides:
+      should_poll
+      async_update
+      async_added_to_hass
+
+    The SwitchEntity class provides the functionality of a ToggleEntity and additional power
+    consumption methods and state attributes.
+    """
+
+    def __init__(self, coordinator, idx):
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator)
+        self.idx = idx
+        self.module = self.coordinator.data[self.idx]
+
+    @property
+    def name(self):
+        """Name of the device."""
+        return self.module.name
+
+    @property
+    def unique_id(self):
+        """ID (unique) of the device."""
+        return self.idx
+
+    @property
+    def device_info(self):
+        """Device information."""
+        return {
+            "identifiers": {
+                # Unique identifiers within the domain
+                (DOMAIN, self.unique_id)
+            },
+            "name": self.name,
+            "manufacturer": "Legrand",
+            "model": HW_TYPE.get(self.module.hw_type),
+            "sw_version": self.module.fw,
+        }
+
+    @property
+    def device_class(self):
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        if self.module.device == "plug":
+            return DEVICE_CLASS_OUTLET
+        return DEVICE_CLASS_SWITCH
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available.
+
+        This is the case when the coordinator is able to update the data successfully
+        AND the switch entity is reachable.
+
+        This method overrides the one of the CoordinatorEntity
+        """
+        return self.coordinator.last_update_success and self.module.reachable
+
+    @property
+    def is_on(self):
+        """Return entity state."""
+        return self.module.status == "on"
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the light on."""
+        # Do the turning on.
+        await self.module.turn_on()
+        # Update the data
+        await self.coordinator.async_request_refresh()
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the entity off."""
+        await self.module.turn_off()
+        # Update the data
+        await self.coordinator.async_request_refresh()
+
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+
+        # Register with the integration's entity map
+        domain = self.registry_entry.platform
+        config_entry_id = self.registry_entry.config_entry_id
+        entity_id = self.registry_entry.entity_id
+        self.hass.data[domain][config_entry_id][ENTITY_UIDS][self.unique_id] = entity_id
```

Comment:
It's enough to keep track of the existing uids. We don't need to know the entity_id. So we can do this accounting from the init module in the coordinator. Update the stored uids before sending the signal to add new entities.

Reply:
OK, I removed the references to the entity_id and now the unique ids are stored in a set. This set is updated before sending the signal to add new entities. Tests have also been adapted to support this new data structure of the uids without entity_ids.